languagetool-core/src/main/java/org/languagetool/UserConfig.java

```diff
@@ -105,38 +109,34 @@ public LinguServices getLinguServices() {
     return linguServices;
   }
 
+  /**
+   * @since 4.3
+   */
+  public String getUserDictName() {
+    return userDictName;
+  }
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
-    UserConfig that = (UserConfig) o;
-    if (maxSpellingSuggestions != that.maxSpellingSuggestions) return false;
-    if (!configurableRuleValues.equals(that.configurableRuleValues)) return false;
-
+    UserConfig other = (UserConfig) o;
     // optimization: equals on userSpecificSpellerWords can be expensive with huge dictionaries
-    if (dictIdentifier != null && that.dictIdentifier != null) {
-      return dictIdentifier.equals(that.dictIdentifier);
-    } else {
-      return userSpecificSpellerWords.equals(that.userSpecificSpellerWords);
-    }
+    // -> we use user id & dictionary names
+    return new EqualsBuilder()
+      .append(configurableRuleValues, other.configurableRuleValues)
+      .append(userDictName, other.userDictName)
+      .append(userSpecificSpellerWords, other.userSpecificSpellerWords)
```

Comment:
This is not in hashCode, is that on purpose? maxSpellingSuggestions is only in hashCode but not here?

Reply:
See comment in hashCode for userSpecificSpellerWords; I simply forgot maxSpellingSuggestions